### PR TITLE
feat: add permission options to create squad modal

### PR DESCRIPTION
**Pull request by capJavert**
### Describe what this PR does
- add permission options to create squad modal
- this one only does the FE work it can be connected once API is done through `WT-1197`
- Minor adjustments to modal design/copy
- Note: currently showing permissions form fields only in `createMode`, in another PR I will continue implementation for squad settings

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1198 #done

### CODE REVIEW

**Comment by sshanzel** [comment]
We will have to update the receiving mutation to accept this parameter, we will have to make changes to the database structure too.

File: packages/shared/src/components/squads/Details.tsx [hunk 1]
```diff
@@ -170,6 +171,28 @@ export function SquadDetails({
...
+            <h4 className="mb-2 font-bold typo-headline text-theme-label-primary">
+              Post permissions
+            </h4>
+            <p className="mb-6 text-salt-90 typo-callout">
+              Choose who is allowed to post new content in this Squad.
+            </p>
+            <RadioItem
+              name="permissionLevel"
+              value="all-members"
+              checked={false}
```
Comment (rebelchris) [inline packages/shared/src/components/squads/Details.tsx:184]: This is not needed right?

File: packages/shared/src/components/squads/Details.tsx [hunk 2]
```diff
@@ -170,6 +171,28 @@ export function SquadDetails({
               }}
             />
           )}
+          <div className="flex flex-col justify-start">
+            <h4 className="mb-2 font-bold typo-headline text-theme-label-primary">
+              Post permissions
+            </h4>
+            <p className="mb-6 text-salt-90 typo-callout">
```
Comment (rebelchris) [inline packages/shared/src/components/squads/Details.tsx:178]: Ah we shouldn't use color versions directly, did linting not complain about it?
And margin-bottom shows as 16 to the box so 4.

```suggestion
            <p className="mb-4 text-theme-label-tertiary typo-callout">
```

File: packages/shared/src/components/squads/Details.tsx [hunk 3]
```diff
@@ -170,6 +171,28 @@ export function SquadDetails({
...
+              checked={false}
+            >
+              All members
+            </RadioItem>
+            <RadioItem
+              name="permissionLevel"
+              value="only-moderators"
+              checked={false}
+            >
+              Only Moderators
```
Comment (rebelchris) [inline packages/shared/src/components/squads/Details.tsx:193]: I think this is a design issue we don't use secondary capitalization unless it's a "brand" like Squad.

File: packages/shared/src/components/squads/Details.tsx [hunk 4]
```diff
@@ -170,6 +171,28 @@ export function SquadDetails({
...
+            <RadioItem
+              name="permissionLevel"
+              value="all-members"
+              checked={false}
+            >
+              All members
+            </RadioItem>
+            <RadioItem
+              name="permissionLevel"
+              value="only-moderators"
```
Comment (rebelchris) [inline packages/shared/src/components/squads/Details.tsx:190]: I think @sshanzel suggested making this a boolean for instance as `restrictMemberPosting`.
So with `onlyModerators` it could be "true" and for all member "false"

Think it makes sense, let me know what you think.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice, have some minor suggestions.

File: packages/shared/src/components/squads/Details.tsx (hunk 2 above)
Comment (capJavert) [reply packages/shared/src/components/squads/Details.tsx:178]: Nope, I did see it sometimes complains. Will keep this in mind but still little hard to follow this colors since they are not directly referenced in design.

**Review by sshanzel — COMMENTED** [review]
Nothing more I can add! Amazing work here! 🚢 🚢 🚢

**Comment by capJavert** [comment]
Just general comment, I think we really need a form library/standardisation, `RadioItem` with boolean shenanigans and also whole mess with `form` and separate states in `Details` are really error prone.

File: packages/shared/src/components/squads/Details.tsx [hunk 5]
```diff
@@ -170,6 +182,30 @@ export function SquadDetails({
...
           )}
+          {createMode && (
+            <div className="flex flex-col justify-start">
+              <h4 className="mb-2 font-bold typo-headline text-theme-label-primary">
+                Post permissions
+              </h4>
+              <p className="mb-4 text-theme-label-tertiary typo-callout">
+                Choose who is allowed to post new content in this Squad.
+              </p>
+              <RadioItem
```
Comment (rebelchris) [inline packages/shared/src/components/squads/Details.tsx:193]: Probably favor using `<Radio>` component, this wraps the items and you can simply pass an array of options to it.

File: packages/shared/src/graphql/sources.ts [hunk 6]
```diff
@@ -46,6 +46,7 @@ export interface Squad extends Source {
   members?: Connection<SourceMember>;
   membersCount: number;
   description: string;
+  memberPostingRank: number;
```
Comment (rebelchris) [inline packages/shared/src/graphql/sources.ts:49]: Should this not be `memberPostingRole`?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Two minor comments, but way to go

File: packages/shared/src/graphql/sources.ts (hunk 6 above)
Comment (rebelchris) [reply packages/shared/src/graphql/sources.ts:49]: Yeah probably we can do that to uniform the edit function, but let's see once we tackle that one.

**Review by rebelchris — APPROVED** [review]
Nice one!

**Comment by capJavert** [comment]
Merging so I can continue with `WT-1196`
